test/test_pipeline/components/test_training.py

```diff
@@ -226,79 +232,140 @@ def test_epoch_training(self):
+    # so the set/get params methods should recreate the same object
+    for name, trainer in trainer_choice.get_components().items():
+        config = trainer.get_hyperparameter_search_space().sample_configuration()
+        estimator = trainer(**config)
+        estimator_clone = clone(estimator)
+        estimator_clone_params = estimator_clone.get_params()
+
+        # Make sure all keys are copied properly
+        for k, v in estimator.get_params().items():
+            assert k in estimator_clone_params
+
+        # Make sure the params getter of estimator are honored
+        klass = estimator.__class__
+        new_object_params = estimator.get_params(deep=False)
+        for name, param in new_object_params.items():
+            new_object_params[name] = clone(param, safe=False)
+        new_object = klass(**new_object_params)
+        params_set = new_object.get_params(deep=False)
+
+        for name in new_object_params:
+            param1 = new_object_params[name]
+            param2 = params_set[name]
+            assert param1 == param2
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    ("tabular_classification", set(['RowCutMixTrainer', 'RowCutOutTrainer'])),
+    ("image_classification", set(['GridCutMixTrainer', 'GridCutOutTrainer'])),
+    ("time_series_classification", set([])),
+])
+def test_get_set_config_space(test_input, expected):
+    """Make sure that we can setup a valid choice in the trainer
+    choice"""
+    trainer_choice = TrainerChoice(dataset_properties={'task_type': test_input})
+    cs = trainer_choice.get_hyperparameter_search_space()
+
+    # Make sure that all hyperparameters are part of the serach space
+    # Filtering out the ones not supported for the given task
+    always_expected_trainers = set(['StandardTrainer', 'MixUpTrainer'])
+    assert set(cs.get_hyperparameter('__choice__').choices) == always_expected_trainers | expected
+
+    # Make sure we can properly set some random configs
+    # Whereas just one iteration will make sure the algorithm works,
+    # doing five iterations increase the confidence. We will be able to
+    # catch component specific crashes
+    for i in range(5):
+        config = cs.sample_configuration()
+        config_dict = copy.deepcopy(config.get_dictionary())
+        trainer_choice.set_hyperparameters(config)
+
+        assert trainer_choice.choice.__class__ == trainer_choice.get_components(
+        )[config_dict['__choice__']]
+
+        # Then check the choice configuration
+        selected_choice = config_dict.pop('__choice__', None)
+        for key, value in config_dict.items():
+            # Remove the selected_choice string from the parameter
+            # so we can query in the object for it
+            key = key.replace(selected_choice + ':', '')
+            assert key in vars(trainer_choice.choice)
+            assert value == trainer_choice.choice.__dict__[key]
+
+
+@pytest.mark.parametrize("cutmix_prob", [1.0, 0.0])
```

Comment:
are you certain this would generate a number between 0 and 1? I tried it multiple times and always got 1. Maybe it should be sample from 0, 1. Also, as in the implementation of mixup [here](https://github.com/franchuterivera/Auto-PyTorch/blob/a423cc98392dbe84cc22ede049ce226dcbac0825/autoPyTorch/pipeline/components/training/trainer/GridCutMixTrainer.py#L35), it may not always change the features of X depending on the r sampled which may cause the test to fail.

Reply:
Actually, the if you see this [line](https://github.com/automl/Auto-PyTorch/blob/a423cc98392dbe84cc22ede049ce226dcbac0825/autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py#L32) we have a greater than operator. in the extreme cases this will return a value on `[0, 1)`. So if the fixture is zero, we have `0 > 0` which means that we do return the exact value of the criterion always and in the other case , when the fixture is 1, because is an open interval 1 is never returned so this test always return `1/fraction (criterion) + (1-1/fraction)`.

So the expectation is that when we have 0 as fixture we return an integer always (as the criterion is a count) or we return a float as we return a fraction of the criterion (which in this case is 0.5* 1 +0.5 * 1 = 1.0). The value is not the important thing but  the object type. Does this make sense? 

Or maybe I am missing something -- in this case, can you elaborate?

Reply:
okay, I see, then it's fine. but currently, the cutout probability fixture will only be 0 and 1 don't you wanna sample from 0, 1?

Reply:
I think the result from sampling is whether you don't do a mixup (probability 0) or you do a mixup (probability 1), so this cases should be covered.

My main motivation not to sample but pass the extreme cutoff probabilities is to make the testcase deterministic, while still testing the possibilities.

Reply:
okay that makes sense.